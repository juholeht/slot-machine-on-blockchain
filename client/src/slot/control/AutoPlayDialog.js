import { React, useState } from 'react';
import { defineMessage, useIntl, injectIntl } from 'react-intl';

import PropTypes from 'prop-types';

import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';

import DialogTitle from '@mui/material/DialogTitle';
import Dialog from '@mui/material/Dialog';
import Button from '@mui/material/Button';

import { STAKE_SIZE_OPTIONS, ETH_CURRENCY_CHAR, PREDEFINED_ROUND_COUNTS } from '../constant';

const INITIAL_ROUNDS = 10;

const AUTO_DIALOG_MESSAGES = {
  stakeLabel: defineMessage({
    id: 'stakeLabel',
    description: 'Label for choosing the stake',
    defaultMessage: 'Stake',
  }),
  setAutoPlayTitle: defineMessage({
    id: 'setAutoPlayTitle',
    description: 'Title for setting the autoplay',
    defaultMessage: 'Set autoplay',
  }),
  totalRoundsLabel: defineMessage({
    id: 'totalRoundsLabel',
    description: 'Label for total rounds selected',
    defaultMessage: 'Total rounds',
  }),
  startButton: defineMessage({
    id: 'startButton',
    description: 'Text for start button',
    defaultMessage: 'Start',
  }),
};

export const AutoPlayDialog = ({
  dialogOpen, onCloseCallback, onAutoDialogStart, stakeIndex, onStakeChanged,
}) => {
  const [rounds, setRounds] = useState(INITIAL_ROUNDS);
  const intl = useIntl();

  const onStart = () => {
    onAutoDialogStart(rounds);
    setRounds(INITIAL_ROUNDS);
    onCloseCallback();
  };

  return (
    <Dialog onClose={onCloseCallback} aria-labelledby="simple-dialog-title" open={dialogOpen}>
      <DialogTitle id="simple-dialog-title">{intl.formatMessage(AUTO_DIALOG_MESSAGES.setAutoPlayTitle)}</DialogTitle>
      <FormControl fullWidth>
        <InputLabel id="stake-select-label">{intl.formatMessage(AUTO_DIALOG_MESSAGES.stakeLabel)}</InputLabel>
        <Select
          key={stakeIndex}
          labelId="stake-select-label"
          id="stake-select"
          value={stakeIndex}
          label={intl.formatMessage(AUTO_DIALOG_MESSAGES.stakeLabel)}
          onChange={({ target }) => onStakeChanged(target.value)}
        >
          {STAKE_SIZE_OPTIONS.reduce((stakeItemArray, item, index) => [...stakeItemArray,
            <MenuItem key={item} value={index}>
              {`${item} ${ETH_CURRENCY_CHAR}`}
            </MenuItem>], [])}
        </Select>
      </FormControl>
      <br />
      <FormControl fullWidth>
        <InputLabel id="total-rounds-select-label">Total rounds</InputLabel>
        <Select
          id="total-rounds-select"
          value={rounds}
          label={intl.formatMessage(AUTO_DIALOG_MESSAGES.totalRoundsLabel)}
          onChange={({ target }) => setRounds(target.value)}
        >
          {PREDEFINED_ROUND_COUNTS.reduce((roundsArray, item) => [...roundsArray,
            <MenuItem key={item} value={item}>
              {item}
            </MenuItem>], [])}
        </Select>
      </FormControl>
      <Button variant="outlined" color="primary" onClick={() => onStart()}>
        {intl.formatMessage(AUTO_DIALOG_MESSAGES.startButton)}
      </Button>
    </Dialog>
  );
};

AutoPlayDialog.propTypes = {
  dialogOpen: PropTypes.bool.isRequired,
  onCloseCallback: PropTypes.func.isRequired,
  onAutoDialogStart: PropTypes.func.isRequired,
  stakeIndex: PropTypes.number.isRequired,
  onStakeChanged: PropTypes.func.isRequired,
};

export default injectIntl(AutoPlayDialog);
