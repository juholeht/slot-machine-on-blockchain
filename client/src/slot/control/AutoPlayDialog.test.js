import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { AutoPlayDialog } from './AutoPlayDialog';

configure({ adapter: new Adapter() });

jest.mock('react-intl', () => {
  const reactIntl = jest.requireActual('react-intl');
  const intl = reactIntl.createIntl({
    locale: 'en',
  });
  return {
    ...reactIntl,
    useIntl: () => intl,
  };
});

describe('<AutoPlayDialog />', () => {
  it('renders properly with default values', () => {
    const wrapper = shallow(<AutoPlayDialog
      dialogOpen
      onCloseCallback={() => console.log('onCloseCallback')}
      onAutoDialogStart={() => console.log('onAutoDialogStart')}
      stakeIndex={0}
      onStakeChanged={() => console.log('onStakeChanged')}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
