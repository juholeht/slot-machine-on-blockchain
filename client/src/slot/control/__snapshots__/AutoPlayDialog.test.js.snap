// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoPlayDialog /> renders properly with default values 1`] = `
<ForwardRef(Dialog)
  aria-labelledby="simple-dialog-title"
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)
    id="simple-dialog-title"
  >
    Set autoplay
  </ForwardRef(DialogTitle)>
  <ForwardRef(FormControl)
    fullWidth={true}
  >
    <ForwardRef(InputLabel)
      id="stake-select-label"
    >
      Stake
    </ForwardRef(InputLabel)>
    <ForwardRef(Select)
      id="stake-select"
      key="0"
      label="Stake"
      labelId="stake-select-label"
      onChange={[Function]}
      value={0}
    >
      <ForwardRef(MenuItem)
        key="0.0001"
        value={0}
      >
        0.0001 Ξ
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="0.001"
        value={1}
      >
        0.001 Ξ
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="0.01"
        value={2}
      >
        0.01 Ξ
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="0.1"
        value={3}
      >
        0.1 Ξ
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="1"
        value={4}
      >
        1 Ξ
      </ForwardRef(MenuItem)>
    </ForwardRef(Select)>
  </ForwardRef(FormControl)>
  <br />
  <ForwardRef(FormControl)
    fullWidth={true}
  >
    <ForwardRef(InputLabel)
      id="total-rounds-select-label"
    >
      Total rounds
    </ForwardRef(InputLabel)>
    <ForwardRef(Select)
      id="total-rounds-select"
      label="Total rounds"
      onChange={[Function]}
      value={10}
    >
      <ForwardRef(MenuItem)
        key="10"
        value={10}
      >
        10
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="20"
        value={20}
      >
        20
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="50"
        value={50}
      >
        50
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        key="100"
        value={100}
      >
        100
      </ForwardRef(MenuItem)>
    </ForwardRef(Select)>
  </ForwardRef(FormControl)>
  <ForwardRef(Button)
    color="primary"
    onClick={[Function]}
    variant="outlined"
  >
    Start
  </ForwardRef(Button)>
</ForwardRef(Dialog)>
`;
